### PR TITLE
Add get thread method

### PR DESCRIPTION
**Pull request by lily-law**

### CODE REVIEW

File: src/db/utils/get-thread-by-id/get-thread-by-id.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { ThreadModel } from "../../../models/thread/thread.model";
+/**
+ * Finds user by ID and only returns relevant properties
+ * @param userId ObjectId
+ */
+export async function getThreadById({threadId}: {threadId: string}) {
+  const result = await ThreadModel.findById(threadId);
+  return result;
```
Comment (davideastmond) [inline src/db/utils/get-thread-by-id/get-thread-by-id.ts:8]: You could just return await here, no? 
```suggestion
  return await ThreadModel.findById(threadId);
```

File: src/db/utils/get-thread-by-id/get-thread-by-id.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { ThreadModel } from "../../../models/thread/thread.model";
+/**
+ * Finds user by ID and only returns relevant properties
+ * @param userId ObjectId
```
Comment (davideastmond) [inline src/db/utils/get-thread-by-id/get-thread-by-id.ts:4]: For this doc string, should it be `threadId`?
```suggestion
 * @param threadId ObjectId
```

File: src/db/utils/get-thread-by-id/get-thread-by-id.ts (hunk 1 above)
Comment (davideastmond) [inline src/db/utils/get-thread-by-id/get-thread-by-id.ts:8]: I'm wondering. Can we just not use the `ThreadModel.findById` static method directly? Maybe there is a reason why we need a wrapper method?

**Comment by davideastmond** [comment]
Looks good! I just had a few comments.

**Review by davideastmond — COMMENTED** [review]
Just had a few comments.

File: src/db/utils/get-thread-by-id/get-thread-by-id.ts (hunk 2 above)
Comment (lily-law) [reply src/db/utils/get-thread-by-id/get-thread-by-id.ts:4]: :pray:

File: src/db/utils/get-thread-by-id/get-thread-by-id.ts (hunk 1 above)
Comment (lily-law) [reply src/db/utils/get-thread-by-id/get-thread-by-id.ts]: So as it is, yes that makes sense, though I've a draft 114 (114-refactor though not strictly just a refactor now) I'd still like to merge in at some point, which for this does a little more so the client can do a little less... 
`export async function getThreadById({threadId, reqUserId}: {threadId: string, reqUserId: string}): Promise<IThreadResponse> {
  const result = await ThreadModel.findById(threadId);
  if (!result) throw new Error(`Unable to find thread for id: ${threadId}`);
  const currentUserReactions: any = {};
  const reactionsAsArray = Object.entries(result.reactions);
  reactionsAsArray.forEach(([key, value]) => {
    if (value.postedByUserId.toString() === reqUserId) {
      currentUserReactions[key] = value;
    }
  })
  return {
    id: result._id.toString(),
    postedByUserId: result.postedByUserId.toString(),
    visibility: result.visibility,
    content: result.content,
    comments: Object.values(result.comments).sort((a, b) => b.updatedAt.valueOf() - a.updatedAt.valueOf()),
    reactions: result.reactions,
    reactionsCount: reactionsAsArray.length,
    currentUserReactions,
    forks: result.forks,
    aForkOfThreadId: result.aForkOfThreadId,
    updatedAt: result.updatedAt.valueOf().toString(),
    createdAt: result.createdAt.valueOf().toString(),
  };
}`

Comment (davideastmond) [reply src/db/utils/get-thread-by-id/get-thread-by-id.ts]: ahh ok! Makes sense!

**Review by davideastmond — APPROVED** [review]

**Pull request merged**